board_client-sim/sim/hardware_sim.py

```
@@ -0,0 +1,118 @@
+    def clear(self):
+        for x in range(WIDTH):
+            for y in range(HEIGHT):
+                self.buffer[x][y] = (0, 0, 0)
+    def get_bounds(self):
+        return (WIDTH, HEIGHT)
+    def set_font(self, font): pass
+    def measure_text(self, text, scale=1): return len(text) * 8 * scale
+    def cleanup_text_cache(self): pass
+    def text(self, text, x, y, wordwrap=-1, scale=1):
+        # Simple text rendering: just draw each char as a colored block
+        for i, char in enumerate(text):
```

Ruff (0.11.9)

Loop control variable `char` not used within loop body

Rename unused `char` to `_char`

(B007)